core/src/apps/cardano/sign_tx.py

```diff
@@ -40,62 +44,42 @@ def is_change(output, inputs):
 
 
 async def show_tx(
-    ctx,
-    outputs: list,
-    outcoins: list,
+    ctx: wire.Context,
+    keychain: seed.Keychain,
+    inputs: List[CardanoTxInputType],
+    outputs: List[CardanoTxOutputType],
     fee: int,
-    network_name: str,
-    raw_inputs: list,
-    raw_outputs: list,
+    protocol_magic: int,
 ) -> None:
-    for index, output in enumerate(outputs):
-        if is_change(raw_outputs[index].address_n, raw_inputs):
+    total_amount = 0
+    for output in outputs:
+        if should_hide_output(output.address_n, inputs):
             continue
 
-        await confirm_sending(ctx, outcoins[index], output)
+        total_amount += output.amount
 
-    total_amount = sum(outcoins)
-    await confirm_transaction(ctx, total_amount, fee, network_name)
+        if not output.address:
+            address, _ = derive_address_and_node(keychain, output.address_n)
+        else:
+            address = output.address
 
+        await confirm_sending(ctx, output.amount, address)
 
-async def request_transaction(ctx, tx_req: CardanoTxRequest, index: int):
-    tx_req.tx_index = index
-    return await ctx.call(tx_req, CardanoTxAck)
+    await confirm_transaction(ctx, total_amount, fee, protocol_magic)
 
 
 @seed.with_keychain
-async def sign_tx(ctx, msg, keychain: seed.Keychain):
-    progress.init(msg.transactions_count, "Loading data")
-
+async def sign_tx(
+    ctx: wire.Context, msg: CardanoSignTx, keychain: seed.Keychain
+) -> CardanoSignedTx:
     try:
-        attested = len(msg.inputs) * [False]
-        input_coins_sum = 0
-        # request transactions
-        tx_req = CardanoTxRequest()
-
-        for index in range(msg.transactions_count):
-            progress.advance()
-            tx_ack = await request_transaction(ctx, tx_req, index)
-            tx_hash = hashlib.blake2b(
-                data=bytes(tx_ack.transaction), outlen=32
-            ).digest()
-            tx_decoded = cbor.decode(tx_ack.transaction)
-            for i, input in enumerate(msg.inputs):
-                if not attested[i] and input.prev_hash == tx_hash:
-                    attested[i] = True
-                    outputs = tx_decoded[1]
-                    amount = outputs[input.prev_index][1]
-                    input_coins_sum += amount
-
-        if not all(attested):
-            raise wire.ProcessError(
-                "No tx data sent for input " + str(attested.index(False))
-            )
-
         transaction = Transaction(
-            msg.inputs, msg.outputs, keychain, msg.protocol_magic, input_coins_sum
+            keychain, msg.inputs, msg.outputs, msg.protocol_magic, msg.fee, msg.ttl,
```

Comment:
as proposed offline, this class indeed seems redundant if it's used just to serialize the tx and we can just keep the `serialise_tx` call below. Btw, I noticed, other coins use the term `serialize`, not `serialise` so it may be worth staying consistent for better searchability